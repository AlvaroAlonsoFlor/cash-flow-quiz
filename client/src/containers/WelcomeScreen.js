import React, {Component, Fragment} from 'react';
import NameForm from '../components/NameForm';
import { CHANGE_POSITION } from '../constants/action-types';


export default class WelcomeScreen extends Component {

    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this)
    }

    handleClick() {
       
        if (window.store.getState().playerName.trim() !== "") {
            window.store.dispatch({ type: CHANGE_POSITION })
            const position = window.store.getState().questionPosition  
            this.props.history.push(`/questions/${position}`)                       
        }     
        
    }

    render() {
        
        return(
            <Fragment>
                <h1>Welcome to the Cash Flow Quiz!</h1>
                <h3>What is your name?</h3>               
                <NameForm />
                <button onClick={this.handleClick}>Next</button>
                          
            </Fragment>
        )
    }
}
