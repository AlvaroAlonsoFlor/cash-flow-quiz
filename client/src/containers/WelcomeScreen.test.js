import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import WelcomeScreen from './WelcomeScreen.js'

configure({adapter: new Adapter()});

describe('WelcomeScreen renders', () => { 
    it('WelcomeScreen should render correctly with no props', () => {
        const component = shallow( < WelcomeScreen / > );

        expect(component).toMatchSnapshot();
    });
});