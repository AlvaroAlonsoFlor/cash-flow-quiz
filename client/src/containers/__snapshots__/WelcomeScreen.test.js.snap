// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders WelcomeScreen should render correctly with no props 1`] = `
<Fragment>
  <h1>
    Welcome to the Cash Flow Quiz!
  </h1>
  <h3>
    What is your name?
  </h3>
  <NameForm />
  <button
    onClick={[Function]}
  >
    Next
  </button>
</Fragment>
`;
